src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -73,13 +93,22 @@ static void HandleNodeBrowse(void * context, DnssdService * services, size_t ser
         discoveryContext->Retain();
         // For some platforms browsed services are already resolved, so verify if resolve is really needed or call resolve callback
 
-        auto & ipAddress = services[i].mAddress;
+        auto & ipAddress         = services[i].mAddress;
+        bool isOperationalBrowse = (strncmp(services[i].mType, kOperationalServiceName, sizeof(kOperationalServiceName)) == 0 &&
+                                    strlen(services[i].mType) == strlen(kOperationalServiceName));
 
-        // Check if SRV, TXT and AAAA records were received in DNS responses
-        if (strlen(services[i].mHostName) == 0 || services[i].mTextEntrySize == 0 || !ipAddress.has_value())
+        // if SRV, TXT and AAAA records were received in DNS responses, also for operational browse result we currently
+        // don't need IP address hence skip resolution.
+        if (!isOperationalBrowse &&
```

Comment:
You can make the `if (isOperationalBrowse)` the first one to avoid checking if `!isOperationalBrowse`.